installer/Makefile

```diff
@@ -53,7 +54,13 @@ vendor:
 	go mod verify
 
 only-build:
+ifeq ($(UNAME),Darwin)
+	echo Darwin
+	make binary
```

Comment:
In Mac, `make binary` is called and it calls `autogen` and `vendor` prerequisites, on the other hand, In Linux, this target calls `$(MAKE) shell ...` which doesn't call  `autogen` and `vendor` prereqiusites.
So, this is not same action, I think.

How about change [`shell` target](https://github.com/kubernetes-sigs/cluster-addons/blob/af88a3421fa5edcad9666e9a8eefe91298cb3612/installer/Makefile#L12-L23) to add `GOOS` and `GOARCH` on `docker run` in Mac?
When specify `GOOS=darwin` and `GOARCH=amd64`, go compiles binary for Mac.